server/service/orbit_client.go

```diff
@@ -211,9 +213,29 @@ func (oc *OrbitClient) enrollAndWriteNodeKeyFile() (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("enroll request: %w", err)
 	}
-	if err := os.WriteFile(oc.nodeKeyFilePath, []byte(orbitNodeKey), constant.DefaultFileMode); err != nil {
-		return "", fmt.Errorf("write orbit node key file: %w", err)
+
+	if runtime.GOOS == "windows" {
+
+		// creating the secret file with empty content
+		if err := os.WriteFile(oc.nodeKeyFilePath, nil, constant.DefaultFileMode); err != nil {
+			return "", fmt.Errorf("create orbit node key file: %w", err)
+		}
+
+		// restricting file access
+		if err := platform.ChmodRestrictFile(oc.nodeKeyFilePath); err != nil {
+			return "", fmt.Errorf("apply ACLs: %w", err)
+		}
+
+		// writing raw key material to the acl-ready secret file
+		if err := os.WriteFile(oc.nodeKeyFilePath, []byte(orbitNodeKey), constant.DefaultFileMode); err != nil {
+			return "", fmt.Errorf("write orbit node key file: %w", err)
+		}
```

Comment:
Nit: This is executed on both branches.

Reply:
good catch! I've just fixed this on the code